common/src/node.rs

```diff
@@ -167,3 +443,262 @@ fn populate_node(graph: &mut Graph, node: NodeId) {
         chunks: Chunks::default(),
     });
 }
+
+/// Represents a particular axis in a voxel grid.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum CoordAxis {
+    X = 0,
+    Y = 1,
+    Z = 2,
+}
+
+impl CoordAxis {
+    /// Iterates through the the axes in ascending order
+    pub fn iter() -> impl ExactSizeIterator<Item = Self> {
+        [Self::X, Self::Y, Self::Z].iter().copied()
+    }
+
+    /// Returns the pair axes orthogonal to the current axis
+    pub fn other_axes(self) -> [Self; 2] {
+        match self {
+            Self::X => [Self::Y, Self::Z],
+            Self::Y => [Self::Z, Self::X],
+            Self::Z => [Self::X, Self::Y],
+        }
+    }
+}
+
+impl From<usize> for CoordAxis {
```

Comment:
I went ahead with `TryFrom`, using the `anyhow::Error` type. This seems idiomatic for non-library projects in Rust, but given the ad-hoc nature of anyhow, I'm not certain, and there might be a better error type to use. To avoid unnecessary code churn, I would like to know your thoughts before I merge this PR.

Reply:
Resolving this conversation, as it's been moved to a new thread.